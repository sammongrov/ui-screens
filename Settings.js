import React, { Component } from 'react';
import { Share, View, ScrollView, TouchableOpacity } from 'react-native';
import { styles } from 'react-native-theme';
import PropTypes from 'prop-types';
import { NavBar, Text, Screen, Icon, Button } from '@ui/components';

export default class Settings extends Component {
  constructor(props) {
    super(props);
    this.state = {
      publicKey: '',
      mnemonic: '',
    };
  }

  componentWillMount = async () => {
    const { loadWallet } = this.props;
    const wallet = await loadWallet();
    this.setState({
      publicKey: wallet.publicKey,
      mnemonic: wallet.mnemonic,
    });
  };

  render = () => {
    const { publicKey, mnemonic } = this.state;
    return (
      <Screen>
        <NavBar
          leftComponent={
            <TouchableOpacity onPress={this.onBackPress}>
              <Icon name={this.navIconName} color={this.navIconColor} size={this.navIconSize} />
            </TouchableOpacity>
          }
          titleText="Settings"
        />
        <ScrollView contentContainerStyle={styles.scrollContentContainer}>
          <View style={[styles.centerColumnContainer, styles.padding15]}>
            <View style={styles.centerContainer}>
              <View>
                <Text style={styles.privateKeyTitle}>Public Key</Text>
                <Text style={styles.clientId}>{publicKey}</Text>
              </View>
              <View>
                <Text style={styles.privateKeyTitle}>Recovery phrase</Text>
                <Text style={styles.clientId}>{mnemonic}</Text>
              </View>
              <View style={styles.buttonContainer}>
                <Button
                  title="Export"
                  onPress={() => {
                    Share.share({
                      message: `Public Key: ${publicKey} // Recovery phrase: ${mnemonic}`,
                      title: 'Wallet sec',
                    });
                  }}
                  buttonStyle={[styles.btnPrimaryBg, styles.margin5]}
                />
              </View>
            </View>
          </View>
        </ScrollView>
      </Screen>
    );
  };
}

Settings.propTypes = {
  loadWallet: PropTypes.func.isRequired,
};

Settings.defaultProps = {};
