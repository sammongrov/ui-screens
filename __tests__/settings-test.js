import React from 'react';
import renderer from 'react-test-renderer';
import { Share } from 'react-native';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Alert } from 'react-native';
import Settings from '../Settings';

configure({ adapter: new Adapter() });

jest.mock('Share', () => {
  const share = {
    share: jest.fn(),
  };
  return share;
});

const loadWallet = () => Alert.alert('Help');

it('About renders correctly', () => {
  const tree = renderer.create(<Settings loadWallet={loadWallet} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Settings button is clicked', () => {
  const rootComponent = shallow(<Settings />);
  const firstButton = rootComponent.find('Button').first();
  firstButton.props().onPress();
  expect(Share.share).toBeCalled();
});
