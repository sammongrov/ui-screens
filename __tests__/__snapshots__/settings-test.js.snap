// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            Settings
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <RCTScrollView>
        <View>
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View>
              <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      null,
                    ]
                  }
                >
                  Public Key
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      null,
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      null,
                    ]
                  }
                >
                  Recovery phrase
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      null,
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View>
                <View
                  style={
                    Array [
                      Object {},
                      false,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#2089dc",
                            "borderRadius": 3,
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          Array [
                            undefined,
                            undefined,
                          ],
                          false,
                          false,
                          false,
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "color": "white",
                              "fontSize": 18,
                              "padding": 8,
                              "textAlign": "center",
                            },
                            undefined,
                            false,
                            false,
                          ]
                        }
                      >
                        Export
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
